pkg/matcher/annotation_matcher.go

```diff
@@ -149,7 +150,16 @@ func getName(prun *tektonv1.PipelineRun) string {
 	return name
 }
 
-func MatchPipelinerunByAnnotation(ctx context.Context, logger *zap.SugaredLogger, pruns []*tektonv1.PipelineRun, cs *params.Run, event *info.Event, vcx provider.Interface) ([]Match, error) {
+func appendAnnotationIfNotEmpty(annotationKey, annotationValue string) []string {
+	annotations := []string{}
+	if annotationValue != "" {
+		annotations = append(annotations, annotationKey)
```

Comment:
i am not sure i understand that function? what does it do, why does it need to append ? i can understand the value of the function of that function but you can re-implement it in a more simple way, isnt it ? or perhaps it's not very well named and that makes it confusing?

Reply:
Yeah, thanks for the review. It does the same thing as the one which you suggested, but yes the name of the function sounds a bit confusing and also the one which you suggested seems to be a bit more simpler. Will update the PR, thanks

>and have that function called in the for _, prun := range pruns loop with:
Yeah, adding it in the for loop but inside this condition 

```golang
if celExpr, ok := prun.GetObjectMeta().GetAnnotations()[keys.OnCelExpression]; ok 
```
as this will check if it has cel expression and then will check for annotations